### PR TITLE
Implemented cards

### PR DESCRIPTION
**Pull request by ThomasLerner**
Implemented Dark Sphere and Lodestone Bauble.

Can't remove garbage commits from pull for some reason, but altered files look innocent.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/l/LodestoneBauble.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    @Override
+    public LodestoneBaubleEffect copy() {
+        return new LodestoneBaubleEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = null;
+        boolean countered = false;
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/l/LodestoneBauble.java]: What's the purpose of this?

File: Mage.Sets/src/mage/cards/l/LodestoneBauble.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        UUID targetId = this.getTargetPointer().getFirst(game, source);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/l/LodestoneBauble.java]: From the gatherer rulings:
>“Up to four” means zero, one, two, three, or four. If you choose zero targets, you still choose a player. That player will draw a card at the beginning of the next turn’s upkeep.

So if nothing is targeted, a player still needs to be chosen. It's probably why this card wasn't already implemented.

File: Mage.Sets/src/mage/cards/l/LodestoneBauble.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        if (targetId != null) {
+            controller = game.getPlayer(game.getControllerId(targetId));
+        }
+        if (targetId != null
+                && game.getStack().counter(targetId, source.getSourceId(), game)) {
+            countered = true;
+        }
+        if (controller != null) {
+            Effect effect = new DrawCardTargetEffect(new StaticValue(1), false);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/l/LodestoneBauble.java]: You can just use `           Effect effect = new DrawCardTargetEffect(1);`

File: Mage.Sets/src/mage/cards/l/LodestoneBauble.java (hunk 1 above)
Comment (ThomasLerner) [reply Mage.Sets/src/mage/cards/l/LodestoneBauble.java]: Ah, that would be unnecessary fluff left over from trying to replicate the way Arcane Denial implements its delayed-draw trigger. Removed it.

File: Mage.Sets/src/mage/cards/l/LodestoneBauble.java (hunk 2 above)
Comment (ThomasLerner) [reply Mage.Sets/src/mage/cards/l/LodestoneBauble.java]: This is interesting. I think I can do a player choice during the resolution of the ability if no target is selected, but it's a unique implementation so I don't know if that's a good practice.

**Comment by theelk801** [comment]
I think I'd try to use the implementation of Drafna's Restoration, which is a card I've just discovered due to its oracle text update. You can just set the target player's nottarget to true.

**Comment by ThomasLerner** [comment]
I'm trying to rewrite Lodestone Bauble in a way that mirrors the current implementation of Drafna's Restoration, but I'm running into tons of nullpointer exceptions due to the specific way it implements the targets and because it's a sorcery rather than an activated ability.

My previous commit fixed the issue in another way, but I can't quite get it to work if I use this method.

**Comment by theelk801** [comment]
The issue is that the player needs to be chosen when you activate the ability, not when it resolves. It's definitely a pain in the neck, that's for sure.

**Pull request closed without merging**